### DISCUSSION TITLE
Behaviour when verify=true option for remote

### DISCUSSION THREAD

**Discussion by jbogar** [treeverse/dvc#9086] [Help]
What exactly is the behavior when verify=true for a remote?

If I understand the documentation (https://dvc.org/doc/command-reference/remote/modify#available-parameters-for-all-remotes), it works like this:
 - download the data
 - calculate the hash
 - compare to hash in the .dvc file (or dvc.lock or wherever)
 - ?
 
 What is the next step? Does it throw error about corrupted data or does it try to download again? I assume an error is thrown?

**Comment by efiop** [comment]
> compare to hash in the .dvc file (or dvc.lock or wherever)

@jbogar Not really, it checks whether the object name matches its hash. So if object is called `123456` it checks that the hash is `123456`
